### PR TITLE
Eliminate unnecessary cache pool resize

### PR DESCRIPTION
**Pull request by gza060625**
1) Implement one more null check after thread entering the monitor to avoid
unnecessary resize cache pool if pool has been resized by other threads
entered the monitor before current thread.

2) Remove pool resize after Tenure Expansion
Copy-cache pool size should be a function of Nursery size, not Tenure size. Removing resizing from `collectorExpanded()` avoid unnecessary pool expansions.

3) Rename varialbe result => resizePerformed for readability.

Signed-off-by: Enson Guo <enson.guo@ibm.com>

### CODE REVIEW

File: gc/base/standard/Scavenger.cpp [hunk 1]
```diff
@@ -3036,8 +3036,12 @@ MM_Scavenger::getFreeCache(MM_EnvironmentStandard *env)
 		OMRPORT_ACCESS_FROM_OMRPORT(env->getPortLibrary());
 		uint64_t duration = omrtime_current_time_millis();
 
+		bool result = false;
 		omrthread_monitor_enter(_freeCacheMonitor);
-		bool result = _scavengeCacheFreeList.resizeCacheEntries(env, 1+_scavengeCacheFreeList.getAllocatedCacheCount(), 0);
+		cache=_scavengeCacheFreeList.popCache(env);
+		if(NULL == cache){
```
Comment (RSalman) [inline gc/base/standard/Scavenger.cpp:3042]: Space after `if` and after closing parentheses

File: gc/base/standard/Scavenger.cpp [hunk 2]
```diff
@@ -3036,8 +3036,12 @@ MM_Scavenger::getFreeCache(MM_EnvironmentStandard *env)
 		OMRPORT_ACCESS_FROM_OMRPORT(env->getPortLibrary());
 		uint64_t duration = omrtime_current_time_millis();
 
+		bool result = false;
 		omrthread_monitor_enter(_freeCacheMonitor);
-		bool result = _scavengeCacheFreeList.resizeCacheEntries(env, 1+_scavengeCacheFreeList.getAllocatedCacheCount(), 0);
+		cache=_scavengeCacheFreeList.popCache(env);
```
Comment (RSalman) [inline gc/base/standard/Scavenger.cpp:3041]: Space after before/after assignment `=`

File: gc/base/standard/Scavenger.cpp [hunk 3]
```diff
@@ -3036,8 +3036,12 @@ MM_Scavenger::getFreeCache(MM_EnvironmentStandard *env)
 		OMRPORT_ACCESS_FROM_OMRPORT(env->getPortLibrary());
 		uint64_t duration = omrtime_current_time_millis();
 
+		bool result = false;
 		omrthread_monitor_enter(_freeCacheMonitor);
-		bool result = _scavengeCacheFreeList.resizeCacheEntries(env, 1+_scavengeCacheFreeList.getAllocatedCacheCount(), 0);
+		cache = _scavengeCacheFreeList.popCache(env);
```
Comment (RSalman) [inline gc/base/standard/Scavenger.cpp:3041]: We should add a comment here to point out why we're reattempting to acquire a cache

**Comment by RSalman** [comment]
This is ready to be merged, @rwy0717 could you please have a look

**Comment by rwy0717** [comment]
@genie-omr build all

**Pull request merged**